### PR TITLE
Add long error explanation for E0578

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #61137

r? @kinnison

### CODE REVIEW

File: src/librustc_resolve/error_codes.rs [hunk 1]
```diff
@@ -1661,6 +1661,30 @@ match eco {
 ```
 "##,
 
+E0578: r##"
+A module cannot be found and therefore, the visibility cannot be determined.
```
Comment (kinnison) [inline src/librustc_resolve/error_codes.rs]: The comma here is odd.  Either 'therefore' is an optional part of the sentence in which case it needs a comma before as well as after, or it's not optional and the comma is unnecessary.

e.g.

> A module cannot be found and, therefore, the visibility cannot be determined.

vs.

> A module cannot be found and therefore the visibility cannot be determined.

**Review by kinnison — CHANGES_REQUESTED** [review]
A comma comment, otherwise :+1:

**Review by Dylan-DPC-zz — APPROVED** [review]

**Review by kinnison — APPROVED** [review]
:+1:

**Comment by GuillaumeGomez** [comment]
Rebased.

**Comment by jonas-schievink** [comment]
> Part of #64903

That looks like the wrong issue

**Comment by GuillaumeGomez** [comment]
@jonas-schievink Indeed, and that's not the first time even though it seemed like I entered the link correctly. This is rather strange... Anyway, updated the link.

**Comment by GuillaumeGomez** [comment]
Rebased.

**Comment by GuillaumeGomez** [comment]
Rebased (and let's hope it's the last time ><).

**Pull request merged**